src/libraries/System.Private.CoreLib/src/System/IO/FileStream.cs

```diff
@@ -78,17 +74,27 @@ private static void ValidateHandle(SafeFileHandle handle, FileAccess access, int
             {
                 throw new ArgumentOutOfRangeException(nameof(access), SR.ArgumentOutOfRange_Enum);
             }
-            else if (bufferSize <= 0)
+            else if (bufferSize < 0)
```

Comment:
The below error message should be updated.  It currently states that a positive value is required.